packages/wonder-blocks-banner/src/components/banner.js

```diff
@@ -0,0 +1,281 @@
+        case "success":
+            return Color.green;
+        case "warning":
+            return Color.gold;
+        case "critical":
+            return Color.red;
+        default:
+            return Color.blue;
+    }
+};
+
+// TODO(WB-1409): Use Phosphor icons instead of custom svgs. Also, use
+// Wonder Blocks Icon instead of img in the render fucntion.
+const iconForKind = (kind: BannerKind) => {
+    switch (kind) {
+        case "info":
+            return "/info.svg";
+        case "success":
+            return "/success.svg";
+        case "warning":
+            return "/warning.svg";
+        case "critical":
+            return "/critical.svg";
+        default:
+            return "/info.svg";
+    }
+};
+
+/**
+ * Banner. A banner displays a prominent message and related optional actions.
+ * It can be used as a way of informing the user of important changes.
+ * Typically, it is displayed toward the top of the screen.
+ *
+ * ### Usage
+ * ```jsx
+ * import Banner from "@khanacademy/wonder-blocks-banner";
+ *
+ * <Banner
+ *     text="Here is some example text."
+ *     kind="success"
+ *     layout="floating"
+ *     actions={[
+ *         {title: "Button 1", onClick: () => {}},
+ *         {title: "Button 2", onClick: () => {}},
+ *     ]}
+ *     onDismiss={() => {console.log("Has been dismissed.")}}
+ * />
+ * ```
+ */
+const Banner = (props: Props): React.Node => {
+    const {actions, onDismiss, kind, layout = "full-width", text} = props;
+    const layoutstyle = {
+        borderRadius: layout && layout === "full-width" ? 0 : 4,
+    };
+
+    const renderActions = () => {
+        return actions?.filter(Boolean).map((action) => {
+            const handleClick = action.onClick;
+            if (action.href) {
+                return (
+                    <View style={styles.action} key={action.title}>
+                        <Link
+                            kind="primary"
+                            href={action.href}
+                            onClick={handleClick}
+                            aria-label={action.ariaLabel ?? action.title}
+                        >
+                            {action.title}
+                        </Link>
+                    </View>
+                );
+            } else if (action.onClick) {
+                return (
+                    <View style={styles.action} key={action.title}>
+                        <Button
+                            kind="tertiary"
+                            size="medium"
+                            aria-label={action.ariaLabel ?? action.title}
+                            onClick={handleClick}
+                        >
+                            {action.title}
+                        </Button>
+                    </View>
+                );
+            } else {
+                throw new Error(
+                    "An action must have an href or an onClick field.",
+                );
+            }
```

Comment:
**suggestion:** Expanding on my previous comment about the `ActionTrigger` type. If we modify the types, we can get rid of this runtime check.